packages/components/src/Combobox/Combobox.tsx

```diff
@@ -1,191 +1,71 @@
-    function handleClickOutside(event: MouseEvent) {
-      if (
-        contentRef.current &&
-        !contentRef.current.contains(event.target as Node)
-      ) {
-        setOpen(false);
-      }
-    }
-
-    document.addEventListener("mousedown", handleClickOutside);
-    return () => {
-      document.removeEventListener("mousedown", handleClickOutside);
-    };
-  }, [contentRef]);
-
-  return (
-    <ComboboxContext.Provider value={{ open, setOpen }}>
-      <div className={styles.combobox} ref={contentRef}>
-        {props.children}
-      </div>
-    </ComboboxContext.Provider>
+export const Combobox = (props: ComboboxProps): JSX.Element => {
+  const { renderContent, renderTrigger } = useComboboxValidation(
+    props.children,
   );
-};
-
-export const Combobox = (props: ComboboxProps) => {
-  return <ComboboxContextProvider>{props.children}</ComboboxContextProvider>;
-};
-
-const TriggerButton = (props: TriggerProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
-
-  return <Button label={props.label} onClick={() => setOpen(!open)} />;
-};
-
-const TriggerChip = (props: TriggerProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
   return (
-    <div
-      className={styles.trigger}
-      onClick={() => {
-        setOpen(!open);
-      }}
-    >
-      <Text>{props.label}</Text>
-      <div className={styles.triggerIcon}>
-        <Icon name="add" size="small" />
-      </div>
-    </div>
+    <ComboboxContextProvider>
+      {renderTrigger}
+      {renderContent}
+    </ComboboxContextProvider>
   );
 };
 
-const Content = (props: ContentProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
-  const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
-
-  const filteredOptions = props.options.filter(option =>
-    option.toLowerCase().includes(searchValue.toLowerCase()),
-  );
-
-  function handleSearch(event: React.ChangeEvent<HTMLInputElement>) {
-    setSearchValue(event.target.value);
-  }
+function useComboboxValidation(children: ReactNode): {
+  renderTrigger: ReactNode;
+  renderContent: ReactNode;
+} {
+  const childrenArray = React.Children.toArray(children);
+  let renderTrigger: ReactNode, renderContent: ReactNode;
+
+  childrenArray.forEach(child => {
+    if (isTriggerElement(child)) {
+      if (renderTrigger) {
+        useAssert(true, "Combobox can only have one Trigger element");
+      }
+      renderTrigger = child;
+    }
+    if (isContentElement(child)) {
+      renderContent = child;
+    }
+  });
 
-  function handleSelection(event: React.MouseEvent<HTMLButtonElement>) {
-    const selected = event.currentTarget.innerText;
-    setSelectedOption(selected);
-    setOpen(false);
-    setSearchValue("");
-    props.onSelection(selected);
+  if (!renderTrigger || !renderContent) {
+    useAssert(true, "Combobox must have a Trigger and Content element");
```

Comment:
Maybe more clearer if we refer to the component name?

```suggestion
    useAssert(true, "Combobox must have a Combobox.Trigger and Combobox.Content children");
```

Reply:
yeah shore! never hurts to be more specific

the only thing is that we don't have a `Combobox.Trigger` the actual component will be either `Combobox.TriggerChip` or `Combobox.TriggerButton` and I'm not sure that we'd want to merge those together to make a single `Trigger` with props, since each one will want pretty different things

Reply:
oh, yeah missed that part. Hmmm maybe not then haha

Reply:
went ahead with half of this suggestion where it'll specify `Combobox.Content` but just `Trigger`

lmk if you think that's better or worse!